include/pybind11/stl.h

```diff
@@ -144,7 +144,7 @@ template <typename Type, typename Value> struct list_caster {
     using value_conv = make_caster<Value>;
 
     bool load(handle src, bool convert) {
-        if (!isinstance<sequence>(src) || isinstance<str>(src))
+        if (!isinstance<sequence>(src) || (!isinstance<bytes>(src) && isinstance<str>(src)))
```

Comment:
I believe this only cures a symptom of a bigger problem, pointed out to me by @bstaletic :
https://github.com/pybind/pybind11/pull/1463/commits/77059f00096b093c1b18223ade7b6fce6a79e819
With that change, the code here can stay as-is.
@wjakob Did you attempt work based on @jagerman 's code, or do you know if anyone else tried?
@cdump While I think it's best to not change this line here, your test looks valuable to me. I'm in favor of merging it after the more fundamental issue is fixed.